teps/0007-conditions-beta.md

```diff
@@ -297,56 +299,88 @@ We can explore adding more [Operators](https://github.com/kubernetes/kubernetes/
+  - an _ordering dependency_: based on `runAfter` which provides sequencing of `Nodes` when there may not be resource dependencies
+- a `Branch` is a `Node` and its dependent `Nodes`
 
-To provide more flexibility when `WhenExpressions` evaluate to `False`, we propose adding a field - `whenSkipped` - that:
-- is used to specify whether to execute `Tasks` that are ordering-dependent on the skipped guarded `Task`
-- defaults to `skipBranch` and users can set it to `runBranch` to allow execution of the rest of the branch
-- can be specified in `Tasks` guarded with `WhenExpressions` only; there will be a validation error if `whenSkipped` is specified in `Tasks` without `WhenExpressions`
-- can take the values `skipBranch` or `runBranch` only; there will be a validation error if anything else is passed to `whenSkipped`
-- can be specified guarded `Tasks` with ordering dependencies only; there will be a validation error if `whenSkipped` is specified in `Tasks` with resource dependencies
+We already support specifying a list of `WhenExpressions` through the `when` field, as described in [Efficiency](#efficiency), with the following syntax:
+
+```yaml
+when:
+  - input: 'foo'
+    operator: in
+    values: ['bar']
+```
+
+To provide more flexibility when `WhenExpressions` evaluate to `False`, we propose adding `expressions` and `scope` fields under the `when` field where:
+- The `expressions` field will contain the list of `WhenExpressions` as described in [Efficiency](#efficiency)
+- The `scope` fields will contain the scope of the `WhenExpressions`, that is whether the `WhenExpressions` guard the `Node` only or the whole `Branch`
 
-Here's how a user can use `whenSkipped` to execute ordering-dependent tasks:
+The syntax would be:
 
 ```yaml
-api: tekton.dev/v1beta1
-kind: Pipeline
-metadata:
-    name: generate-file
-spec:
-  workspaces:
-    - name: source-repo
-  params:
-    - name: path
-      default: 'README.md'
-  tasks:
-    - name: create-file # executed
-      taskRef:
-        name: create-readme-file
-      workspaces:
-        - name: source
-          workspace: source-repo
-    - name: file-exists # executed
-      taskRef:
-        name: file-exists
-      workspaces:
-        - name: source
-          workspace: source-repo
-    - name: echo-file-does-not-exist # skipped
-      when:
+when:
+  scope: Node/Branch
+  expressions:
+    - input: 'foo'
+      operator: in
+      values: ['bar']
+```
+
+The `scope` field is optional, defaults to `Branch` and can be set to `Branch` to specify that the `Node`'s `WhenExpressions` guard the execution of the `Branch` (the `Node` and its dependent `Nodes`). However, when `Node` has ordering dependencies only, users can set the `scope` field to `Node` to specify that the `Node`'s `WhenExpressions` guard the execution of the `Node` only, thus the all dependent nodes (which are ordering dependent tasks only) can be executed. 
```

Comment:
I understand `Node` as `self`, when expressions is limited to that particular pipelineTask/node where it's defined. And `Branch` includes all dependencies i.e. all dependencies are impacted (skipped) by the when expressions of a parent task.